backend/src/server/routes/index.ts

```diff
@@ -839,6 +840,11 @@ export const registerRoutes = async (
     user: userDAL
   });
 
+  // Rate limiters and security headers
+  if (appCfg.isProductionMode) {
+    await server.register<FastifyRateLimitOptions>(ratelimiter, globalRateLimiterCfg());
```

Comment:
How come we decided to move this check?

Reply:
Because before the rate limit was configured before we parsed the auth methods and thier types. Now, we parse the request for auth type before we try to rate limit it

Reply:
This makes sense